Objective-C/CBLReadOnlyDocument.mm

```diff
@@ -109,4 +135,80 @@ - (BOOL) exists {
 }
 
 
+- (id<CBLConflictResolver>) effectiveConflictResolver {
+    return self.database.config.conflictResolver ?: [CBLDefaultConflictResolver new];
+}
+
+
+- (NSData*) encode: (NSError**)outError {
+    // CBLDocument overrides this
+    fleece::slice body = _c4Doc.rawDoc->selectedRev.body;
+    return body ? body.copiedNSData() : [NSData data];
+}
+
+
+#pragma mark - RESOLVING REPLICATED CONFLICTS:
+
+
+- (bool) resolveExistingConflict: (NSError**)outError {
+    // Read the conflicting remote revision:
+    auto otherDoc = [[CBLReadOnlyDocument alloc] initWithDatabase: self.database
+                                                       documentID: self.documentID
+                                                        mustExist: YES
+                                                            error: outError];
+    if (!otherDoc || ![otherDoc selectConflictingRevision])
+        return false;
+
+    // Read the common ancestor revision (if it's available):
+    auto baseDoc = [[CBLReadOnlyDocument alloc] initWithDatabase: self.database
+                                                      documentID: self.documentID
+                                                       mustExist: YES
+                                                           error: outError];
+    if (![baseDoc selectCommonAncestorOfDoc: self andDoc: otherDoc] || !baseDoc.data)
```

Comment:
This is a question. Why do we need to check baseDoc.data? Doesn't -selectCommonAncestorOfDoc return false if there is no base doc?